server/asset/asset.go

```diff
@@ -0,0 +1,33 @@
+package asset
+
+import (
+	"encoding/binary"
+)
+
+const (
+	HashLen  = 32
+	UTXOSize = HashLen + 4
+)
+
+// TODO: This UTXO may be too specific with a fixed HashLen.
+
+type UTXO struct {
+	TxHash [HashLen]byte
+	Vout   uint32
+}
+
+func (u *UTXO) Serialize() []byte {
+	b := make([]byte, UTXOSize)
+	copy(b[:HashLen], u.TxHash[:])
+	binary.LittleEndian.PutUint32(b[HashLen:], u.Vout)
```

Comment:
Without actually implementing too much in the asset package (since you're working on that), I basically need a UTXO data type that can be serialized.  Perhaps what I need is a UTXO interface (defined in the order package) with the Serialize* methods to get the data I need in e.g. `(o *MarketOrder).Serialize`.  Is that what you are suggesting as a solution for the order types?

Reply:
And for testing in isolation from the asset package I can make a stub that satisfies this UTXO interface.